src/utils/document-revision-comparison-utils.js

```diff
@@ -0,0 +1,211 @@
+      return DIFF_TYPE.add;
+    case SECTION_CHANGE_TYPE.removed:
+      return DIFF_TYPE.delete;
+    default:
+      return DIFF_TYPE.modify;
+  }
+}
+
+export function getInterleavedSectionsChanges(oldSectionKeys, newSectionKeys) {
+  const oldSectionKeySet = new Set(oldSectionKeys);
+  const newSectionKeySet = new Set(newSectionKeys);
+  const orderedCommonKeysInOldSection = oldSectionKeys.filter(value => newSectionKeySet.has(value));
+  const orderedCommonKeysInNewSection = newSectionKeys.filter(value => oldSectionKeySet.has(value));
+  const longestCommonSubsequence = mdiff(orderedCommonKeysInOldSection, orderedCommonKeysInNewSection).getLcs();
+  const nonMovedSectionKeySet = new Set(longestCommonSubsequence);
+  const movedSectionKeySet = new Set(orderedCommonKeysInNewSection.filter(value => !nonMovedSectionKeySet.has(value)));
+
+  const result = [];
+  const processedSectionKeys = new Set();
+  let indexInOldSections = 0;
+  let indexInNewSections = 0;
+
+  while (indexInOldSections < oldSectionKeys.length || indexInNewSections < newSectionKeys.length) {
+    const oldSectionKey = oldSectionKeys[indexInOldSections] ?? null;
+    const newSectionKey = newSectionKeys[indexInNewSections] ?? null;
+
+    if (oldSectionKey === newSectionKey) {
+      // section stayed in the same place
+      result.push([oldSectionKey, SECTION_CHANGE_TYPE.unchanged]);
+      processedSectionKeys.add(oldSectionKey);
+      indexInOldSections += 1;
+      indexInNewSections += 1;
+    } else if (newSectionKey !== null && !oldSectionKeySet.has(newSectionKey)) {
+      // section was added in `sectionKeys2`
+      result.push([newSectionKey, SECTION_CHANGE_TYPE.added]);
+      processedSectionKeys.add(newSectionKey);
+      indexInNewSections += 1;
+    } else if (oldSectionKey !== null && !newSectionKeySet.has(oldSectionKey)) {
+      // section was deleted in `sectionKeys2`
+      result.push([oldSectionKey, SECTION_CHANGE_TYPE.removed]);
+      processedSectionKeys.add(oldSectionKey);
+      indexInOldSections += 1;
+    } else if (newSectionKey !== null && movedSectionKeySet.has(newSectionKey)) {
+      // section was moved to here in `sectionKeys2`
+      result.push([newSectionKey, SECTION_CHANGE_TYPE.movedHere]);
+      processedSectionKeys.add(newSectionKey);
+      indexInNewSections += 1;
+    } else if (oldSectionKey !== null && movedSectionKeySet.has(oldSectionKey)) {
+      // section was moved from here to a different place in `sectionKeys2`
+      result.push([oldSectionKey, processedSectionKeys.has(oldSectionKey) ? SECTION_CHANGE_TYPE.movedUp : SECTION_CHANGE_TYPE.movedDown]);
+      processedSectionKeys.add(oldSectionKey);
+      indexInOldSections += 1;
+    } else {
+      throw new Error('Unexpected case while creating section diff');
+    }
+  }
+
+  return result;
+}
+
+function getMetadataTextRepresentation(documentRevision) {
+  if (!documentRevision) {
+    return '';
+  }
+
+  const shallowClone = cloneDeep({ ...documentRevision });
```

Comment:
This is not a shallow clone actually, right? 🤔 And destructuring isn't needed, I think.

Reply:
HA! Seems like I first did the one, then the other, and ended up having both with the name of the other. If you're confused now, don't worry, it was intentional.

Switched to only having `cloneDeep` and the variable named `clonedRevision` (although shallow clone would have sufficed, it's dangerous if code is added later). I also realized when fixing this, that we should revert the user object expansion done by the `ClientDataMapper` for the `publicContext.accreditedEditors` array like we do for `createdBy`.

tldr; Fixed! Thanks for spotting it!